testing/src/test/java/io/stargate/it/http/graphfirst/FilesResourceTest.java

```diff
@@ -49,4 +67,85 @@ public void shouldGetGraphQLDirectivesFile() throws IOException {
     assertThat(responseBody).isNotEmpty();
     assertThat(responseBody).contains("directive @cql_input");
   }
+
+  @Test
+  public void shouldGetFileWithSchema(@TestKeyspace CqlIdentifier keyspaceId) throws IOException {
+    // when get schema file
+    String url = String.format("%s:8080/graphqlv2/files/namespace/%s.graphql", host, keyspaceId);
+    Request getRequest =
+        new Request.Builder()
+            .get()
+            .addHeader("content-type", MediaType.TEXT_PLAIN)
+            .url(url)
+            .build();
+
+    Response getSchemaResponse = getHttpClient().newCall(getRequest).execute();
+
+    // then
+    assertThat(getSchemaResponse.code()).isEqualTo(200);
+    assertThat(getSchemaResponse.header("Content-Disposition"))
+        .contains((String.format("%s-%s.graphql", keyspaceId.toString(), DEPLOYED_SCHEMA_VERSION)));
+
+    String responseBody = getSchemaResponse.body().string();
+    assertThat(responseBody).isNotEmpty();
+    assertThat(responseBody).contains("type User");
+    assertThat(responseBody).contains("type Query");
+  }
+
+  @Test
+  public void shouldGetFileWithSchemaUsingVersionArgument(@TestKeyspace CqlIdentifier keyspaceId)
```

Comment:
Would be good to add a test for namespace that doesn't exist. What happens if I provide a valid keyspace that hasn't been graphqlified like `system`?

Reply:
+1 to add a test, as well as for existing namespace but wrong version.

From a quick look at the code it looks like the DAO returns `null` and that gets encoded as the response. I think a better behavior would be to return a 404.